buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -182,6 +180,34 @@ class BuildPlugin implements Plugin<Project> {
         return javaHome
     }
 
+    private static String findRuntimeJavaHome(Project project) {
+        String java8Home = System.getenv('JAVA_8_HOME')
```

Comment:
I think we should use a more explicit name here, like ES_TEST_JAVA_HOME for a couple reasons:
* I personally have env vars for each version of java, so I can quickly switch to that java version with an alias. I always have this env set, so it would mean for me I would always be testing against java 8.
* This doesn't have to be java 8 specific. The same feature of our build can be used to test against a different jvm like azul or ibm, while still building with the official java 9 jdk we use for releases.

Reply:
I discussed this with @rjernst and after some consideration I agree. We agreed to name the variable `RUNTIME_JAVA_HOME` though.